test/Transforms/R1CSLowering/r1cs_lowering_pass.llzk

```diff
@@ -0,0 +1,204 @@
+// RUN: llzk-opt -I %S -split-input-file -llzk-r1cs-lowering-pass -verify-diagnostics %s | FileCheck --enable-var-scope %s
+
+module attributes {veridise.lang = "llzk"} {
+  // lowers constraints to be at most degree 2 polynomials
+  struct.def @CmpConstraint {
+    function.def @compute(%a: !felt.type, %b: !felt.type, %c: !felt.type) -> !struct.type<@CmpConstraint> {
+      %self = struct.new : !struct.type<@CmpConstraint>
+      function.return %self : !struct.type<@CmpConstraint>
+    }
+
+    function.def @constrain(%self: !struct.type<@CmpConstraint>, %a: !felt.type, %b: !felt.type, %c: !felt.type) {
+      %z = felt.mul %a, %b
+      %0 = felt.mul %a, %c
+      constrain.eq %0, %z : !felt.type
+      function.return
+    }
+  }
+}
+
+// CHECK-LABEL:   struct.def @CmpConstraint {
+// CHECK:           function.def @compute(%[[VAL_0:.*]]: !felt.type, %[[VAL_1:.*]]: !felt.type, %[[VAL_2:.*]]: !felt.type) -> !struct.type<@CmpConstraint> attributes {function.allow_witness} {
```

Comment:
nit: I prefer replacing the `.*` with `[0-9a-zA-Z_\.]+` just to be more specific on the matches